### ISSUE TITLE
Error importing Spell.dbc

### ISSUE THREAD

**Issue by Dokman** [stoneharry/Spell-Editor-GUI-V2#14]
![image](https://cloud.githubusercontent.com/assets/2290904/9149416/2db95c58-3da9-11e5-808c-8246a1e19fc7.png)

**Comment by Dokman** [comment]
and for now is in the same thing

**Comment by stoneharry** [comment]
It is taking a very long time to import the spell DBC?

The speed of this depends on the strength of your connection to the MySQL server and the speed of the actual server. It needs to execute (number_of_spells / 250) big queries.

Cancel that (ignore the warning on corrupting the table - you can manually reset the table with the "Truncate Table" button if we need to).

Try running the program from the same machine as your MySQL server.

Alternatively if you cannot run the program on the same machine because it runs something like Linux: you can set up a MySQL server on localhost, import the Spell.DBC into there, then backup the database as a SQL dump. Transfer and import the database dump on your remote machine.

**Comment by Dokman** [comment]
i am running it in my localhost server

**Comment by Dokman** [comment]
don't import anything right now, i was running 8 hours and doesn't import anything

**Comment by stoneharry** [comment]
Try running the program as administrator. When you open the program again what stage does it go to? "Would you like to import a spell.dbc now?"

**Comment by stoneharry** [comment]
Also are you using your own Spell.dbc? Have you tested with the Spell.dbc provided in the release? Again, try running as administrator. You should see the progress bar appear within 30-60 seconds of starting the import.

**Comment by Dokman** [comment]
i am using our spell.dbc

**Comment by stoneharry** [comment]
Can you upload it for me to test?

**Comment by Dokman** [comment]
tomorrow now i will be busy all day TT

**Comment by Dokman** [comment]
@stoneharry i used your latest release 7z file and doesn't import anything too!

**Comment by stoneharry** [comment]
Try running the program as administrator. When you open the program again what stage does it go to? "Would you like to import a spell.dbc now?"

If you import, does it do anything in under a minute?

**Comment by stoneharry** [comment]
The latest version may fix this. I can't deploy a updated binary file right now but here is a temporary hackfix you can use:

https://dl.dropboxusercontent.com/u/1102355/alastor.zip

Let me know if it does anything.

**Comment by Dokman** [comment]
![image](https://cloud.githubusercontent.com/assets/2290904/9153029/c0014362-3e41-11e5-9217-56b43258c4c5.png)

**Comment by stoneharry** [comment]
Basically error reporting was broken during import. Now that I have fixed error reporting we can see the issue.

There is a bug trying to import your spell.dbc - a value for EffectRealPointsPerLevel1 on one of the spells is not considered valid. I need you to upload your spell.dbc for me so that I can debug if this is an issue on my end or your end.

**Comment by Dokman** [comment]
it's your one

**Comment by stoneharry** [comment]
What MySQL server are you using? What version? Is it a remote server or a local server?

Can you try deleting the table/database from the MySQL server and running the program again so that it recreates the table?

**Comment by Dokman** [comment]
Localhost running a mysql 5.6 community server into the last version of 5.6 and it's localhost

**Comment by stoneharry** [comment]
Thank you, I'll take another look tomorrow it's 2am for me now, I need to sleep. :)

**Comment by Dokman** [comment]
me too hahahaha here is at 3:21 AM

**Comment by stoneharry** [comment]
I see why the error is happening but it doesn't really make a lot of sense for that error to happen on that column.

Can you try this version? https://dl.dropboxusercontent.com/u/1102355/alastor.zip (updated since last linked)

Also delete the MySQL table completely.

DROP TABLE `yourspelltable`;

If you don't know how to drop it then change to a new table name in the config that doesn't exist.

That will re-create the table. I have changed the table so that floats are stored as double's. I hope this fixes the issue since EffectRealPointsPerLevel1 is a float.

**Comment by Dokman** [comment]
yes now is working hahahaha

**Comment by stoneharry** [comment]
The spell.DBC crashes? What do you mean?

**Comment by Dokman** [comment]
the p`rogram crashes when is importing the spell.dbc

**Comment by stoneharry** [comment]
Can you test this new version, if it still crashes a Error.txt should at least be created that will contain the error message.

https://dl.dropboxusercontent.com/u/1102355/Spell%20Editor%20V2.exe

**Comment by Dokman** [comment]
![image](https://cloud.githubusercontent.com/assets/2290904/9170933/8a25156c-3f67-11e5-8dca-3ece95267d5b.png)

**Comment by Dokman** [comment]
i tryed with a official spell.dbc extracted from Patch-esES-3.mpq and the same issue

**Comment by stoneharry** [comment]
I really do not understand what could be causing that error on that column, or why it is only happening for you. I'm doing some more research.

**Comment by Dokman** [comment]
i dunno i watched that row it have a 0x0 or 0, like you want to tell i don't know what's the problem

**Comment by Dokman** [comment]
@stoneharry there is any news about the bug?

**Comment by stoneharry** [comment]
I've been busy today.

**Comment by Dokman** [comment]
ok no problem hahaha

**Comment by stoneharry** [comment]
I don't know if this version works any better: https://github.com/stoneharry/Spell-Editor-GUI-V2/releases/tag/v1.4.1

I fixed a bug I was having... I still have no idea on the issues you are having...

**Comment by Haramman** [comment]
I, too, am having this issue.
I'm running:
Windows 7.
MySQL 5.6 with HeidiSQL.
Tried using both the provided DBC along with my own, problem persists.
I've tried truncating the table.
Before this update, I used the last one in which it "failed to connect" to the MySQL with the correct login several times - but upon a restart of the program would ask to import Spell.dbc and then never finish (like the other guy).

**Comment by stoneharry** [comment]
Found the issue, will fix it shortly. Also thank you for making me aware it is not using the correct database/table configuration.

**Comment by stoneharry** [comment]
I'm relatively confident that https://github.com/stoneharry/Spell-Editor-GUI-V2/releases/tag/v1.4.2 will fix the issues.

You need to switch to a new table or execute the query:

`DROP TABLE spell;

This is to allow the table to be recreated with the correct schema. The config bug is now fixed too.

**Comment by Haramman** [comment]
A new bug has appeared; https://gyazo.com/7479dec21d275a4b68bbb43af8b18386
I dropped the table "spell" before using this.

**Comment by Haramman** [comment]
Problem persist, after having tried to dump even databases, changing DB and table along and compiling it myself.

**Comment by Dokman** [comment]
same error, if you change to float i have the same error that afer i had

**Comment by stoneharry** [comment]
Sorry, I made a typo. I'm not really sure how. I'm quite tired today.

Fixed: https://dl.dropboxusercontent.com/u/1102355/Spell%20Editor%20V2.exe

**Comment by Dokman** [comment]
Same error again
![image](https://cloud.githubusercontent.com/assets/2290904/9208039/881c2fd4-4071-11e5-8930-f3268ae0b262.png)

**Comment by Haramman** [comment]
https://gyazo.com/5578788175a8270601cc93539e88ee86
I took have this issue now, however, it now appears to be a different column "EffectMultipleValue1" at row 1 instead of "EffectRealPointsPerLevel1" at row 6.

**Comment by stoneharry** [comment]
I'll have to check what the range of values possible for that column are tomorrow. (20, 10) should probably work instead of (10, 10) if you know how to try that yourself. Otherwise I'll deploy it tomorrow.

**Comment by Dokman** [comment]
how i edit it? i what file is

**Comment by stoneharry** [comment]
I've updated the existing latest release. Hopefully that should fix this error once and for all. Again you need to drop any existing tables.

**Comment by Haramman** [comment]
https://gyazo.com/54c44b02cb7f18b570bf9cecee6e32b8
Still same issue, I remembered to drop the table (and entire database) before starting the program.

**Comment by stoneharry** [comment]
Oh come on!

I've set it so it can have 10 digits then a decimal point then another 10 digits.

e.g:

1000000000.0000000001

That should be more than enough data. Yet it is saying the column is too small to fit the value in.

**Comment by Haramman** [comment]
Is this just Dokman and I who have this issue or do you also have it?

**Comment by Haramman** [comment]
Doesn't appear to work with my own DBC either, trying everything right now. :)

**Comment by stoneharry** [comment]
I don't have it using MySQL Server 5.6 x64. I don't really understand why it is happening for you two on your configurations.

According to the MySQL standard this issue makes sense to occur if the data cannot fit. But I don't understand why the data doesn't fit - it should in my mind.

I need someone more experienced with MySQL to look at this issue. I might ask on StackOverflow.

The minimum value for this column is: -175
The maximum value for this column is: 63034

**Comment by stoneharry** [comment]
I mean looking at the data, NONE of these values are decimal values. It could be error'ing because of this - maybe it is expecting a double and receiving a integer.

**Comment by Dokman** [comment]
So it's fixed?

**Comment by stoneharry** [comment]
No, not yet. I wish it was.

**Comment by stoneharry** [comment]
-Might- be fixed with this version: https://dl.dropboxusercontent.com/u/1102355/SpellEditor_TestVersion.zip

I am now attempting to cast the data during insertion.

**Comment by Haramman** [comment]
https://gyazo.com/85b1af91f91619ea4ef4b775a94fef5f
Different error now, and as you can see in the background, stuff was actually loaded this time.

**Comment by stoneharry** [comment]
This is getting ridiculous. What I assume has happened is the data failed to cast to a double so was inserted as NULL, so when the spell selection takes place it gets NULL which it never should and fails.

For reference, this is how I am now handling the cast on insertion:

                                case TypeCode.Single:
                                    {
                                        q.Append(String.Format("cast('{0}' as decimal(11, 3)), ", f.GetValue(r.record)));
                                        break;
                                    }

**Comment by Haramman** [comment]
I won't pretend to know anything about scripting in general, so this code means nothing to me, sorry.
When do you reckon a new build will be available?

**Comment by stoneharry** [comment]
I'll make a new build tomorrow that prints every query that it is executing so that I can see exactly what is going on.

**Comment by Dokman** [comment]
any news

**Comment by Freddan962** [comment]
I'm experiencing the same issues above.

Windows 8.1
Fresh spell.dbc extraction

**Comment by stoneharry** [comment]
The new release build should fix these errors: https://github.com/stoneharry/Spell-Editor-GUI-V2/releases/tag/v1.4.3

This was tested by somebody and confirmed as fixing the issue. Please let me know if it persists for any of you.

**Comment by stoneharry** [comment]
@Dokman did you drop the table before attempting to use the new version?

**Comment by Freddan962** [comment]
I tested the editor with Jamey and can confirm it solved the issue for me.

I was running MySQL server 5.6 with HeidiSQL

**Comment by Dokman** [comment]
![image](https://cloud.githubusercontent.com/assets/2290904/9289410/c961064a-436e-11e5-8f71-9dd797d1994d.png)

Test it from zero

**Comment by Dokman** [comment]
@stoneharry Any news?

**Comment by stoneharry** [comment]
Not yet. I'm extremely busy this week.

**Comment by Haramman** [comment]
Having the same issue, it imports up and until the spell it has issues with, however.

**Comment by Dokman** [comment]
Thanks for all the support, when you can try to completely fix the error?

**Comment by stoneharry** [comment]
I do not know when it will be fixed. I don't know what is causing the issue and since I cannot reproduce it, it becomes incredibly hard to debug. I do not have a lot of free time currently.

You can always use an older version that is not the MySQL update. Some stuff is broken but it works for the most part.

I might come back to this project in a few weeks.

**Comment by Dokman** [comment]
i am waiting for news when you can...

**Comment by stoneharry** [comment]
I won't look at this for a while, see my previous reply.

**Comment by Dokman** [comment]
![image](https://cloud.githubusercontent.com/assets/2290904/9707613/f458bf54-550d-11e5-8445-e0080cdba265.png)

**Comment by Chasityh** [comment]
I get this error http://prntscr.com/8ekl18

**Comment by Dokman** [comment]
SpellDescription is too short

**Comment by Dokman** [comment]
Any news? @stoneharry

**Comment by stoneharry** [comment]
Not been on the computer at all really, busy with real life. The buggier non MySQL version can be used as a workaround for people with issues on this version.

**Comment by Dokman** [comment]
@stoneharry  what version is that one?

**Comment by stoneharry** [comment]
Release Build 1.3.5

**Comment by madman1851** [comment]
I've got this error "Unknown column 'SpellName0' in 'field list'" . Any help please?

**Comment by Rochet2** [comment]
Solved SQL errors with:
https://gist.github.com/Rochet2/f8d5757bfd7bd1daadf0

One issue was that the text fields are too small to contain data. So tested changing them to TEXT instead.
Another issue is that there is a value like `'1E+17'` that fails to cast to decimal due to it's size. Solved it by removing casts.
Then after that there was an issue that with removing casts I found another issue regarding rounding - Some values are for example `'0,1'` which has a comma instead of  a dot so the casts round them (data trunkated). Without the cast however this was an error and thus I solved it by replacing all commas in decimal values to dots.
